src/window_identifier/mod.rs

```diff
@@ -310,6 +314,47 @@ impl Serialize for WindowIdentifierType {
     }
 }
 
+impl WindowIdentifierType {
+    /// Sets the given window as a modal child of the window represented by
+    /// self. Currently only Wayland windows are supported.
+    ///
+    /// This is useful in backend implementations as the portal dialogs have to
+    /// be modal and grouped together with the application that launched the
+    /// request.
+    ///
+    /// Realizes the window if it is not realized yet.
+    ///
+    /// Returns `true` on success.
+    #[allow(unused_variables)]
+    #[cfg(any(feature = "gtk4_wayland", feature = "gtk4_x11"))]
+    pub fn set_parent_of(&self, window: &gtk::Window) -> bool {
+        let fake_parent = gtk::Window::new();
+        gtk::prelude::GtkWindowExt::set_transient_for(window, Some(&fake_parent));
+        gtk::prelude::GtkWindowExt::set_modal(window, true);
+
+        if gtk::prelude::NativeExt::surface(window).is_none() {
+            gtk::prelude::NativeExt::realize(window);
+        }
+
+        let surface = gtk::prelude::NativeExt::surface(window).unwrap();
+
+        match self {
+            #[cfg(feature = "gtk4_x11")]
+            WindowIdentifierType::X11(_) => {
+                // TODO
+                false
```

Comment:
This https://github.com/flatpak/xdg-desktop-portal-gtk/blob/main/src/externalwindow-x11.c#L48-L110 can help for implementing the X11 side

Reply:
Thanks done.

So:
- Wayland application -> Wayland server = works fine.
- Wayland application -> X11 server = ends in warning as expected: can not cast X11Toplevel to WaylandToplevel.
- X11 application -> X11 server = GTK says it is unable to find the surface corresponding to the XID, I don't understand what I'm doing wrong :thinking:. 
- X11 application -> Wayland server = ends in warning as expected (I think?): can not find surface corresponding to XID.

If we fix the X11 application to X11 server functionality, then it is all set I think.

Reply:
>Wayland application -> X11 server = ends in warning as expected: can not cast X11Toplevel to WaylandToplevel.
Can this even happen?

Reply:
Yep, launch the server under X11 (`GDK_BACKEND=x11`) and application under Wayland. 

So now the portal dialog is of X11Toplevel type and so can not be cast to a WaylandToplevel. So it fails with warning here - https://github.com/bilelmoussaoui/ashpd/pull/288/files#diff-54d5d9106ea94967f399a2c21ee27948ae23aa5b39116a93bdabbefb691ece18R394